src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -309,6 +309,42 @@ TransactionManager::submit_transaction_direct(
     trim_alloc_to);
 }
 
+TransactionManager::update_lba_mappings_ret
+TransactionManager::update_lba_mappings(
+  Transaction &t)
+{
+  LOG_PREFIX(TransactionManager::update_lba_mappings);
+  SUBTRACET(seastore_t, "update extent lba mappings", t);
+  return seastar::do_with(
+    std::list<LogicalCachedExtentRef>(),
+    std::list<CachedExtentRef>(),
+    [this, &t](auto &lextents, auto &pextents) {
+    t.for_all_fresh_blocks([&lextents, &pextents](auto &extents) {
+      for (auto &extent : extents) {
+        if (extent->is_valid()) {
+          if (extent->is_logical()) {
+            if (!extent->get_last_committed_crc()) {
+              extent->set_last_committed_crc(extent->get_crc32c());
+            }
+            assert(extent->get_crc32c() == extent->get_last_committed_crc());
+            lextents.emplace_back(extent->template cast<LogicalCachedExtent>());
+          } else {
+            pextents.emplace_back(extent);
+          }
+        }
+      }
+    });
+    return lba_manager->update_mappings(
+      t, lextents
+    ).si_then([&pextents] {
+      for (auto &extent : pextents) {
+        assert(!extent->is_logical() && extent->is_valid());
+        extent->set_last_committed_crc(extent->get_crc32c());
```

Comment:
For logical extents, the crc is stored in the parent lba entry, and for physical extents, the crc is stored inside the extent buffer, they are logical counterparts.

To be consistent with logical extent crc mgmts, I think it would be better not to mix the common `last_committed_crc` logic with the physical extent crc mgmts.